src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(1L);
+        user.setLogin("rightLogin");
+        user.setPassword("rightPassword");
+        user.setAge(25);
+    }
+
+    @AfterEach
+    void tearDown() {
+        ((StorageDaoImpl) storageDao).clear();
+    }
+
+    @Test
+    void register_rightUser_ok() {
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+
+        User user2 = new User();
+        user2.setId(2L);
+        user2.setLogin("123456");
+        user2.setPassword("123456");
+        user2.setAge(18);
+        User actual2 = registrationService.register(user2);
+        assertEquals(user2, actual2);
+    }
+
+    @Test
+    void register_userWithExistLogin_notOk() {
+        Storage.people.add(user);
+        User userWithExistLogin = user;
```

Comment:
You should create a new user instance with the same login instead of reusing the existing 'user' object. This would ensure that the test is not affected by the state of the 'user' object, keeping the test cases independent.